tiny_dnn/core/kernels/conv2d_op_internal.h

```diff
@@ -58,54 +58,59 @@ inline void conv2d_op_internal(const tensor_t &in_data,
-        const float_t *pw = &W[idx];
-
-        idx               = params.in_padded.get_index(0, 0, inc);
-        const float_t *pi = &in[idx];
-
-        idx         = params.out.get_index(0, 0, o);
-        float_t *pa = &a[idx];
-
-        for (serial_size_t y = 0; y < params.out.height_; y++) {
-          for (serial_size_t x = 0; x < params.out.width_; x++) {
-            const float_t *ppw = pw;
-            const float_t *ppi =
-              pi + params.in_padded.width_ * (y * params.h_stride) +
-              x * params.w_stride;
-            float_t sum{0};
-
-            // should be optimized for small kernel(3x3,5x5)
-            for (serial_size_t wy = 0; wy < params.weight.height_;
-                 wy++) {  // NOLINT
-              for (serial_size_t wx = 0; wx < params.weight.width_;
-                   wx++) {  // NOLINT
-                idx = wy * params.in_padded.width_ + wx;
-                sum += *ppw++ * ppi[idx];
-              }
-            }
-            pa[y * params.out.width_ + x] += sum;
-          }
-        }
-      }
-
-      if (params.has_bias) {
-        float_t *pa  = &a[params.out.get_index(0, 0, o)];
-        float_t *paa = pa + params.out.width_ * params.out.height_;
-        std::for_each(pa, paa, [&](float_t &f) { f += bias[o]; });
-      }
-    }
-  });
+  for_(parallelize, 0, in_data.size(),
+       [&](const blocked_range &r) {
+         size_t out_area         = params.out.area();
+         serial_size_t iw        = params.in_padded.width_;
+         serial_size_t id        = params.in.depth_;
+         serial_size_t ow        = params.out.width_;
+         serial_size_t oh        = params.out.height_;
+         serial_size_t od        = params.out.depth_;
+         serial_size_t kw        = params.weight.width_;
+         serial_size_t kh        = params.weight.height_;
+         serial_size_t w_stride  = params.w_stride;
+         serial_size_t h_stride  = params.h_stride;
+         serial_size_t h_stride2 = iw * h_stride;
+         for (int sample = r.begin(); sample < r.end(); sample++) {
+           const vec_t &in = in_data[sample];
+           vec_t &a        = out_data[sample];
+           for (serial_size_t o = 0; o < od; o++) {
+             float_t *pa = &a[params.out.get_index(0, 0, o)];
+             for (serial_size_t inc = 0; inc < id; inc++) {
+               if (!params.tbl.is_connected(o, inc)) continue;
+               serial_size_t idx;
+               idx               = params.weight.get_index(0, 0, id * o + inc);
+               const float_t *pw = &W[idx];
+               idx               = params.in_padded.get_index(0, 0, inc);
+               const float_t *pi = &in[idx];
+               float_t *pa2      = pa;
+               for (serial_size_t y = 0; y < oh; y++) {
+                 const float_t *pi2 = pi;
+                 for (serial_size_t x = 0; x < ow; x++) {
+                   const float_t *pi3 = pi2;
```

Comment:
Is it possible to somehow improve naming? I'm trying to read the code and getting lost very fast.  A big amount of two-letter variables is confusing.